src/status_im2/contexts/onboarding/events.cljs

```diff
@@ -13,23 +14,28 @@
  (fn [request]
    (status/create-account-and-login request)))
 
-(rf/defn on-delete-profile-success
-  {:events [:onboarding-2/on-delete-profile-success]}
-  [{:keys [db]} key-uid]
-  {:db (update-in db [:multiaccounts/multiaccounts] dissoc key-uid)})
+(re-frame/reg-fx
+ ::validate-mnemonic
+ (fn [[mnemonic on-success on-error]]
+   (status/validate-mnemonic
+    (security/safe-unmask-data mnemonic)
+    (fn [result]
+      (let [{:keys [error]} (types/json->clj result)]
+        (if (seq error)
+          (on-error error)
```

Comment:
Nothing major, but it's common practice in re-frame to create effects that *optionally* take `on-success` and `on-error` callbacks. Hence there's always a check for existence: `(when on-error (on-error error))`. Our `json-rpc/call` effect doesn't do this and we end up with some ugly stuff passing empty anonymous functions `#()`.